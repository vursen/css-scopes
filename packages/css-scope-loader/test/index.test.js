const compiler = require('./compiler');

test('zero config', async () => {
  const stats = await compiler('./fixtures/bundle.js');
  const output = stats.toJson().modules[1].source

  expect(output).toMatchSnapshot()
});

test('custom extensions', async () => {
  const stats = await compiler('./fixtures/bundle.js', {
    extensions: ['.slim']
  });

  const output = stats.toJson().modules[1].source

  expect(output).toMatchSnapshot()
});

test('default scope path', async () => {
  const stats = await compiler('./fixtures/bundle.js', {
    defaultScopePath(sourceName, _sourcePath) {
      return `./${sourceName}.js`
    }
  });

  const output = stats.toJson().modules[1].source

  expect(output).toMatchSnapshot()
});

test('using class as scope selector', async () => {
  const stats = await compiler('./fixtures/bundle.js', {
    scopeSelector(scopePath) {
      return `.s-${scopePath.replace(/\/|\./g, '-')}`;
    }
  });

  const output = stats.toJson().modules[1].source

  expect(output).toMatchSnapshot()
});
