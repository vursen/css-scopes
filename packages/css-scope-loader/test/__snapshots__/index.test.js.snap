// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom extensions 1`] = `
"exports = module.exports = require(\\"../../../node_modules/css-loader/dist/runtime/api.js\\")(false);
// Module
exports.push([module.id, \\"[s-fixtures-components-Component-slim].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-slim].host [s-fixtures-components-Component-slim].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-slim].host [s-fixtures-components-Component-html].host {\\\\n  font-size: 20px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-slim].host [s-fixtures-components-Component-slim].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\", \\"\\"]);
"
`;

exports[`default scope path 1`] = `
"exports = module.exports = require(\\"../../../node_modules/css-loader/dist/runtime/api.js\\")(false);
// Module
exports.push([module.id, \\"[s-fixtures-components-Component-js].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-js].host [s-fixtures-components-Component-html].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-js].host [s-fixtures-components-Component-html].host {\\\\n  font-size: 20px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-js].host [s-fixtures-components-Component-html].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\", \\"\\"]);
"
`;

exports[`using class as scope selector 1`] = `
"exports = module.exports = require(\\"../../../node_modules/css-loader/dist/runtime/api.js\\")(false);
// Module
exports.push([module.id, \\".s-fixtures-components-Component-html.host {\\\\n  font-size: 15px;\\\\n}\\\\n\\\\n.s-fixtures-components-Component-html.host .s-fixtures-components-Component-html.host {\\\\n  font-size: 15px;\\\\n}\\\\n\\\\n.s-fixtures-components-Component-html.host .s-fixtures-components-Component-html.host {\\\\n  font-size: 20px;\\\\n}\\\\n\\\\n.s-fixtures-components-Component-html.host .s-fixtures-components-Component-html.host {\\\\n  font-size: 15px;\\\\n}\\\\n\\", \\"\\"]);
"
`;

exports[`zero config 1`] = `
"exports = module.exports = require(\\"../../../node_modules/css-loader/dist/runtime/api.js\\")(false);
// Module
exports.push([module.id, \\"[s-fixtures-components-Component-html].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-html].host [s-fixtures-components-Component-html].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-html].host [s-fixtures-components-Component-html].host {\\\\n  font-size: 20px;\\\\n}\\\\n\\\\n[s-fixtures-components-Component-html].host [s-fixtures-components-Component-html].host {\\\\n  font-size: 15px;\\\\n}\\\\n\\", \\"\\"]);
"
`;
